### ISSUE TITLE
Always getting User closed login dialog on FirebaseUI 0.3.1

### ISSUE THREAD

**Issue by jaibatrik** [firebase/FirebaseUI-Android#101]
Line 77 of `GoogleAuthProvider.java` is 
```java
((Activity)this.getContext()).startActivityForResult(signInIntent, GoogleActions.SIGN_IN);
```

When I do `showFirebaseLoginPrompt()` and click on the *Login with Google* button, it goes to the above line. However, the `Activity` never shows up. Rather it hits `onActivityResult` with `resultCode` 0 (`RESULT_CANCELED`).

I have enabled Google authentication in my Firebase dashboard and provided the **web** client id and secrets there.

**Comment by jaibatrik** [comment]
Also getting `ConnectionResult{statusCode=INTERNAL_ERROR, resolution=null}` as the `Intent` extra.

**Comment by dazza5000** [comment]
I'm seeing the same thing. Did you find a fix?

**Comment by dazza5000** [comment]
I got Facebook auth working in this tutorial, but didn't get the Google Auth to work using this tutorial, but I did get Google Auth working in this one: https://github.com/firebase/firebase-login-demo-android

**Comment by jaibatrik** [comment]
Yes, [this](http://stackoverflow.com/a/31462183/1086930) is the solution, but I bet it is not clearly documented.

**Comment by puf** [comment]
Great to hear that you found the cause. On the documentation: PRs are most
welcome. :-)

**Comment by Queskr** [comment]
Hey,

Not sure I understood what the solution is :(

Jaibatrik's answer points to:
"In my case, I had to change the Client ID to use the exact package name of the activity class building the GoogleApiClient object, not a higher-level package."

So I need to change my Android OAuth's key in GoogleAPIs' console to point not to the actual Android package but to the Activity which sparked the Fragment of the login process? After that, do I need to change anything in the Firebase configuration (which is now configured with the Google's Web OAuth key/secret)? Do I need to regenerate the config file?

Any clarity would be greatly appreciated - I'm working on this challenge for 4 days now :(

Thanks!!!
Yohay

**Comment by jaibatrik** [comment]
Yes, I only needed to make change in the Google Cloud Console, and not in
the app itself.

On 12 April 2016 at 22:02, Queskr <notifications@github.com> wrote:

> Hey,
>
> Not sure I understood what the solution is :(
>
> Jaibatrik's answer points to:
> "In my case, I had to change the Client ID to use the exact package name
> of the activity class building the GoogleApiClient object, not a
> higher-level package."
>
> So I need to change my Android OAuth's key in GoogleAPIs' console to point
> not to the actual Android package but to the Activity which sparked the
> Fragment of the login process? After that, do I need to change anything in
> the Firebase configuration (which is now configured with the Google's Web
> OAuth key/secret)? Do I need to regenerate the config file?
>
> Any clarity would be greatly appreciated - I'm working on this challenge
> for 4 days now :(
>
> Thanks!!!
> Yohay
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/firebase/FirebaseUI-Android/issues/101#issuecomment-208994112>
>



-- 
with best wishes,
Jaibatrik
www.jaibatrik.info

**Comment by Queskr** [comment]
OK, I've tried it several times. 
- My main Activity which is derived from FirebaseLoginBaseActivity is called QueskrFeedActivity. 
- My package name is com.queskr.www.queskrandroidapp

I've tried setting both the Android Credentials in the Google API Manager and the "Enable Google Services for your App" to com.queskr.www.queskrandroidapp (the package name) and/or com.queskr.www.queskrandroidapp.QueskrFeedActivity, several times. I still get the error.

I'm in deep frustration now. Any step by step walkthrough or a working example I can use will be greatly appreciated. I am willing to pay for assistance at this point.

Anyone who can assist will be greatly appreciated ...
